runtime/vm/vmprops.c

```diff
@@ -1040,13 +1040,22 @@ getJ9VMVersionString(J9JavaVM *vm) {
 UDATA
 addSystemProperty(J9JavaVM * vm, const char* propName,  const char* propValue, UDATA flags)
 {
+	char *convertedPropValue = NULL;
 	J9VMSystemProperty* newProp = pool_newElement(vm->systemProperties);
 	if (NULL == newProp) {
 		return J9SYSPROP_ERROR_OUT_OF_MEMORY;
 	}
 
+	/* Convert the value string from the platform encoding to the modified UTF8 */
+	convertedPropValue = (char *)getMUtf8String(vm, propValue, strlen(propValue));
```

Comment:
This is going to be a little trickier. Sometimes the arguments to addSystemProperty() are already converted to utf8.
See https://github.com/eclipse/openj9/blob/master/runtime/vm/vmprops.c#L856-L861 
and https://github.com/eclipse/openj9/blob/master/runtime/vm/vmprops.c#L907

Either the callers of addSystemProperty() need to pass in a new argument to indicate the value should be converted, or the conversion needs to be done elsewhere before addSystemProperty() is called.

Reply:
As discussed with @pshipton offline, the fix above will be moved to `addPropertiesForOptionWithAssignArg` and all functions in addModularitySystemProperties including `addPropertiesForOptionWithAssignArg` should be fixed in the same way to address all these cases for the platform setting.

Reply:
The code has been updated at https://github.com/eclipse/openj9/commit/30329871ee1a60edf626ffa8e542638851913045. Will compile a build locally with the fix to see whether it still works good as expected.